File: 0048-rotate-image/0048-rotate-image.py
class Solution:
    def rotate(self, matrix: List[List[int]]) -> None:
        # Transpose the matrix
        for i in range(len(matrix)):
            for j in range(i, len(matrix)):
                matrix[j][i], matrix[i][j] = matrix[i][j], matrix[j][i]
        
        # Reverse each row
        for i in range(len(matrix)):
            matrix[i] = matrix[i][::-1]

File: 0048-rotate-image/0048-rotate-image.rs
impl Solution {
    pub fn rotate(matrix: &mut Vec<Vec<i32>>) {
        let n = matrix.len();
        
        // Transpose the matrix
        for i in 0..n {
            for j in i..n {
                let temp = matrix[i][j];
                matrix[i][j] = matrix[j][i];
                matrix[j][i] = temp;
            }
        }
        
        // Reverse each row
        for row in matrix.iter_mut() {
            let mid = n / 2;
            for j in 0..mid {
                let temp = row[j];
                row[j] = row[n - j - 1];
                row[n - j - 1] = temp;
            }
        }
    }
}
